src/components/Wizard/index.tsx

```diff
@@ -66,8 +66,9 @@ export default function Wizard() {
     undefined
   );
   const [previewJson, setPreviewJson] = useState<object | undefined>(undefined);
+  const [storage, setStorage] = useState<object | undefined>(undefined);
```

Comment:
We likely want to set up our own hooks for using localStorage. Something like:


```typescript
function useLocalStorage<T>(
  key: string,
  guard(value: T): value is T,
  defaultValue: T,
): [T, React.Dispatch<React.SetStateAction<T>>] {
  // note:
  // React.Dispatch<React.SetStateAction<T>> is:
  //   (newValue: T) => () | (fn: (oldValue: T) => T) => ()
  // ie you can call setState(value) or setState((old) => new);

  const [parsedValue, setParsedValue] = useState(() => {
    const rawValue = window.localStorage.getItem(key);
    if (rawValue) {
      let value;
      try {
        value = JSON.parse(rawValue);
      } catch (e) {
        return defaultValue;
      }

      if (gaurd(value)) {
        return value;
      }
    }
    return defaultValue;
  });

  const setValue = useCallback((newValue: (T | (oldValue: T) => void) => {
    setParsedValue(oldValue => {
      const value = newValue instanceof Function ? newValue(oldValue) : newValue;
      window.localStorage.setItem(key, JSON.stringify(value));
      return value
    });
  }, [key]);
}
```

We can then build an abstraction on top of this hook, creating a hook to get/set/delete messages:


```typescript
interface SavedMessage {
  template: MessageTemplate; // We can do more type level shenanigans later to ensure this matches up with formData
  formData: WizardFormData;
}
type SavedMessages = Record<string, SavedMessage>;

interface UseSavedMessages {
  messges: SavedMessages;
  saveMessage: (id: string, template: MessageTemplate, formData: WizardFormData) => void;
  deleteMessage: (id: string) => void;
}

function isSavedMessages(value: unknown): value is SavedMessages {
  return true; // TODO
}

function useSavedMessages(): UseSavedMessages {
  const [messages, setMessages] = useLocalStorage(
    "savedMessages", isSavedMessages, {});

  const saveMessage = useCallback((id: string, template: MessageTemplate, formData: WizardFormData) => {
    setMessages(oldMessages => {
      const newMessages = { ...oldMessages };
      newMessages[id] = { template, formData };
      return newMessages;
  }, [setMessages])

  const deleteMessage = useCallback((id: string) => {
    setMessages(oldMessages => {
      const newMessages = { ...oldMessages };
      delete newMessages[id];
      return newMessages;
    });
  }, [setMessages]);

  return { savedMessages, saveMessage, deleteMessage };
}

```